CONTRIBUTING.md

```diff
@@ -0,0 +1,82 @@
+# Contributing to Caliptra
+
+## Contributor License Agreement
+
+Per the [Charter](Caliptra_WG_Technical_Charter.md), contributors to Caliptra
+must have signed the CHIPS CLA.
+
+## Guidelines for code contributions
+
+* Changes must be made via pull request
+* [Branch protection rules](github_rules.md) ensure pull requests are reviewed
+  by project members
+* [Maintainers](MAINTAINERS.md) are expected to enforce `CODEOWNERS` rights
+  where necessary. Maintainers are also the sole owners of the repo's
+  configuration.
+* The branch protection rules include constraints to maintain a linear history
+  and avoid merges in pull requests.
+* Please keep a clean commit history, and no long series
+  of "fixup" patches. Use GitHub's "Squash and merge" option if necessary.
+* New code files must include the Apache license 2.0 header. Commit verification
+  tools enforce this.
+* Fork the repos into your personal repo, and create pull requests from your
+  personal repo. Do not pollute the main repo with your personal branches.
+
+## Design discussion
+
+* All design issues are discussed via GitHub issues.
+* Please reference the issue in your commit where appropriate.
+* Chat channels should be primarily used for short, tactical discussions
+  (example, coordinate resolution of a merge conflict, coordinate update of
+  settings, meeting coordinates)
+* Chat channel participation, however, is restricted to developers who have
+  signed the CHIPS CLA, or are affiliiated with companies that have signed the
+  CHIPS CLA.
+
+## Branch policy
+
+* For all repos there is a single `main` branch.
+* Private dev branches for organizations must integrate back to main
+* TODO: upload the schematic and reference, for RTL branch policy
+* `$major.8` tags must exist in `main`. At that point changes to RTL must follow
+  CCB rules below
+* `$major.9` tags for RTL mark the beginning of bug fixes only period.
+* Specific integrations should be forked off `main` into a release branch at a
+  named tag. Example:
+  - `release/1.2.1` forks off `main` at tag `1.2`.
+  - Linear history in that branch includes tags `1.2.1.0`, `1.2.1.1` and so on
+  - A new integration can branch off `release/1.2.2` at tag `1.2` as well.
+  - A new integration can branch off `release/1.3.1` at tag `1.3` as well
+* Branch protection rules apply to `main` and `release/*` branches.
+* All commits in `release/*` branches must be found in `main` as well. Since bug
+  fixes added to a `release` branch may be merged into `main` later in the
+  history, the `release/` branch commit must include a `Release-Commit=$SHA`
```

Comment:
We can enforce this with branch protection rules combined with a github workflow.

Reply:
Do you propose removing the `Release-Commit` requirement? Identical commits cherry-picked to different branches have different SHA1

Reply:
No. But we can write a check that enforces that all release/ commits include a Release-Commit=$VALID_MAIN_SHA in the message.

Reply:
Issue filed on Caliptra-sw, later we can do it on RTL as well.

https://github.com/chipsalliance/caliptra-sw/issues/48